pyaerocom/aeroval/modelmaps_engine.py

```diff
@@ -176,24 +209,91 @@ def _process_map_var(self, model_name, var, reanalyse_existing):
+        """
+
+        try:
+            data = self.read_gridded_obsdata(model_name, var)
+        except EntryNotAvailable:
+            try:
+                data = self.read_model_data(model_name, var)
+            except Exception as e:
+                raise ModelVarNotAvailable(
+                    f"Cannot read data for model {model_name} (variable {var}): {e}"
+                )
+
+        var_ranges_defaults = self.cfg.var_scale_colmap
+
+        if var in var_ranges_defaults.keys():
+            cmapinfo = var_ranges_defaults[var]
+            varinfo = VarinfoWeb(var, cmap=cmapinfo["colmap"], cmap_bins=cmapinfo["scale"])
         else:
-            lat_res = self.cfg.modelmaps_opts.maps_res_deg["lat_res_deg"]
-            lon_res = self.cfg.modelmaps_opts.maps_res_deg["lon_res_deg"]
+            cmapinfo = var_ranges_defaults["default"]
+            varinfo = VarinfoWeb(var, cmap=cmapinfo["colmap"], cmap_bins=cmapinfo["scale"])
 
-        datajson = griddeddata_to_jsondict(data, lat_res_deg=lat_res, lon_res_deg=lon_res)
+        data = self._check_dimensions(data)
 
-        with self.avdb.lock():
-            self.avdb.put_gridded_map(
-                datajson, self.exp_output.proj_id, self.exp_output.exp_id, var, model_name
-            )
-            self.avdb.put_contour(
-                contourjson, self.exp_output.proj_id, self.exp_output.exp_id, var, model_name
+        outdir = self.cfg.path_manager.get_json_output_dirs()["contour/overlay"]
+
+        maps_freq = TsType(self.cfg.modelmaps_opts.maps_freq)
+
+        if maps_freq == "coarsest":  # TODO: Implement this in terms of a TsType object. #1267
+            freq = min(TsType(fq) for fq in self.cfg.time_cfg.freqs)
+            freq = min(freq, self.cfg.time_cfg.main_freq)
+        else:
+            freq = maps_freq
+        tst = TsType(data.ts_type)
+
+        if tst < freq:
+            raise TemporalResolutionError(f"need {freq} or higher, got{tst}")
+        elif tst > freq:
+            data = data.resample_time(str(freq))
+
+        data.check_unit()
+
+        tst = _jsdate_list(data)
+        data = data.to_xarray()
+        for i, date in enumerate(tst):
+            outname = f"{model_name}_{var}_{date}"
+
+            # Note should matche the output location defined in aerovaldb
+            fp_overlay = os.path.join(outdir, outname)
+
+            if not reanalyse_existing:
+                if os.path.exists(fp_overlay):
+                    logger.info(f"Skipping overlay processing of {outname}: data already exists.")
+                    continue
+
+            overlay_plot = plot_overlay_pixel_maps(
+                data[i],
+                cmap=varinfo.cmap,
+                cmap_bins=varinfo.cmap_bins,
+                format=self.cfg.modelmaps_opts.overlay_save_format,
             )
 
-        return [fp_json, fp_geojson]
+            with self.avdb.lock():
+                self.avdb.put_map_overlay(
+                    overlay_plot,
```

Comment:
Does this mean that put_map_overlay has no knowledge of the overlay_plot image-type (png/jpeg,webp)? And the get_map_overlay does not have that knowledge either?

@thorbjoernl 
I would feel better if the image-type were known..., e.g. 
`ROUTE_MAP_OVERLAY = "/v0/map-overlay/{project}/{experiment}/{source}/{variable}/{date}.{img_type}"` would even indicate that the files on dict in the jsonfiledb are images...